### PR TITLE
Refactor cert CommanderGenerator usage

### PR DESCRIPTION
**Pull request by mfurtak**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description

Carry the work and lessons learned from #8129 over to _cert_

### Motivation and Context

_cert_ does not currently have any option conflicts, but this refactoring is still generally more correct and safe.

### CODE REVIEW

File: cert/spec/commands_generator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      # launch takes no params, but we want to expect the call and prevent
+      # actual execution of the method
+      expect(runner).to receive(:launch)
+
+      Cert::CommandsGenerator.start
+
+      expect(Cert.config[:username]).to eq('me@it.com')
+    end
+
+    it "platform flag from tool options can be used" do
```
Comment (ohwutup) [inline cert/spec/commands_generator_spec.rb]: Could we rephrase this to be something like `it "can use platform flag from tool options"` to make it read a little nicer or is that confusing the phrase?

File: cert/spec/commands_generator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      expect(runner).to receive(:launch)
+
+      Cert::CommandsGenerator.start
+
+      expect(Cert.config[:platform]).to eq('macos')
+    end
+  end
+
+  describe ":revoke_expired option handling" do
+    it "development flag from tool options can be used" do
```
Comment (ohwutup) [inline cert/spec/commands_generator_spec.rb]: Similar to the above comment, can this be `it "can use development flag from tool options"`?

File: cert/spec/commands_generator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      # revoke_expired_certs! takes no params, but we want to expect the call
+      # and prevent actual execution of the method
+      expect(runner).to receive(:revoke_expired_certs!)
+
+      Cert::CommandsGenerator.start
+
+      expect(Cert.config[:development]).to be(true)
+    end
+
+    it "output_path short flag from tool options can be used" do
```
Comment (ohwutup) [inline cert/spec/commands_generator_spec.rb]: `it "can use output_path short flag from tool options"`?

**Review by ohayon — APPROVED** [review]
This looks great as is, so I approved it, just a couple questions about phrasing of the tests. 👍

**Comment by mfurtak** [comment]
Sounds good!

**Comment by ohwutup** [comment]
💯
